utils.go

```diff
@@ -0,0 +1,55 @@
+package dpop
+
+import (
+	"crypto"
+	"encoding/base64"
+)
+
+// Internal function used to ensure hash is available and set the default
+// to SHA256
+func ValidateHashFunction(hashFn crypto.Hash) crypto.Hash {
+	if hashFn.Available() {
+		return hashFn
+	}
+	return crypto.SHA256
+}
+
+type HashInput struct {
+	inString string
+	hashFn	 crypto.Hash
+}
+
+func (h *HashInput) New(inString string) *HashInput {
+	return &HashInput{inString, crypto.SHA256,}
+}
+
+func (h *HashInput) SetHashFn(hashFn crypto.Hash) {
+	h.hashFn = hashFn
+}
```

Comment:
There is no need to have a constructor or get/set operators as the struct is exported.